File: C/求球反弹高度.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
int main(int argc,char** argv) 
{
	FILE *fp;
	if((fp=fopen("result.txt","w"))==NULL)
		return -1;
	int i,n;
	double h,a[100]={0},x;
	h=atoi(argv[1]);
	n=atoi(argv[2]);
	a[0]=h;
	a[1]=0.5*h;
	x=h;
	for(i=1;i<n;i++)
	{
		a[i+1]=0.5*a[i];
		x+=a[i-1];
	}
	fprintf(fp,"%.2lf %.2lf",x,a[i]);
	fclose(fp);
	return 0;
}

File: Python/求球反弹高度.py
import sys
h=int(sys.argv[1])
n=int(sys.argv[2])
a,x=[],h
a.append(h)
a.append(0.5*h)
for i in range(1,n):
	a.append(0.5*a[i])
	x+=a[i-1]
with open('result.txt','w') as fp:
	x=round(x,2)
	y=round(a[n],2)
	fp.write(str(x)+' '+str(y))